LGTM, 

```
[ERROR] Failed to execute goal com.mycila.maven-license-plugin:maven-license-plugin:1.10.b1:check (default) on project elasticsearch-core: Some files do not have the expected license header -> [Help 1]
```

I think you can run `mvn license:update-file-header` to auto update them and `mvn license:check-file-header` to verify that the build will pass

--- a/README.md
+++ b/README.md
@@ -19,25 +19,25 @@ The library components are offered a la carte:
     <dependency>
       <groupId>com.sumologic.elasticsearch</groupId>
       <artifactId>elasticsearch-core</artifactId>
-      <version>1.0.16</version>
+      <version>1.0.17</version>
     </dependency>
 
     <dependency>
       <groupId>com.sumologic.elasticsearch</groupId>
       <artifactId>elasticsearch-aws</artifactId>
-      <version>1.0.16</version>
+      <version>1.0.17</version>
     </dependency>
 
     <dependency>
       <groupId>com.sumologic.elasticsearch</groupId>
       <artifactId>elasticsearch-akka</artifactId>
-      <version>1.0.16</version>
+      <version>1.0.17</version>
     </dependency>
 
     <dependency>
       <groupId>com.sumologic.elasticsearch</groupId>
       <artifactId>elasticsearch-test</artifactId>
-      <version>1.0.16</version>
+      <version>1.0.17</version>
       <scope>test</scope>
     </dependency>
   ```

--- a/elasticsearch-akka/pom.xml
+++ b/elasticsearch-akka/pom.xml
@@ -4,12 +4,12 @@
   <artifactId>elasticsearch-akka</artifactId>
   <name>elasticsearch-akka</name>
   <packaging>jar</packaging>
-  <version>1.0.16-SNAPSHOT</version>
+  <version>1.0.17-SNAPSHOT</version>
 
   <parent>
     <artifactId>all</artifactId>
     <groupId>com.sumologic.elasticsearch</groupId>
-    <version>1.0.16-SNAPSHOT</version>
+    <version>1.0.17-SNAPSHOT</version>
     <relativePath>../</relativePath>
   </parent>
 
@@ -22,7 +22,7 @@
     <dependency>
       <groupId>com.sumologic.elasticsearch</groupId>
       <artifactId>elasticsearch-core</artifactId>
-      <version>1.0.16-SNAPSHOT</version>
+      <version>1.0.17-SNAPSHOT</version>
     </dependency>
 
     <dependency>

--- a/elasticsearch-aws/pom.xml
+++ b/elasticsearch-aws/pom.xml
@@ -4,13 +4,13 @@
   <artifactId>elasticsearch-aws</artifactId>
   <name>elasticsearch-aws</name>
   <packaging>jar</packaging>
-  <version>1.0.16-SNAPSHOT</version>
+  <version>1.0.17-SNAPSHOT</version>
   <scm><connection>scm:git:git@github.com:Sanyaku/sumologic.git/tags/all-1.0.4/sumologic.git</connection></scm>
 
   <parent>
     <artifactId>all</artifactId>
     <groupId>com.sumologic.elasticsearch</groupId>
-    <version>1.0.16-SNAPSHOT</version>
+    <version>1.0.17-SNAPSHOT</version>
     <relativePath>../</relativePath>
   </parent>
 
@@ -24,7 +24,7 @@
     <dependency>
       <groupId>com.sumologic.elasticsearch</groupId>
       <artifactId>elasticsearch-core</artifactId>
-      <version>1.0.16-SNAPSHOT</version>
+      <version>1.0.17-SNAPSHOT</version>
     </dependency>
 
     <dependency>

--- a/elasticsearch-core/pom.xml
+++ b/elasticsearch-core/pom.xml
@@ -3,12 +3,12 @@
   <artifactId>elasticsearch-core</artifactId>
   <name>elasticsearch-core</name>
   <packaging>jar</packaging>
-  <version>1.0.16-SNAPSHOT</version>
+  <version>1.0.17-SNAPSHOT</version>
 
   <parent>
     <artifactId>all</artifactId>
     <groupId>com.sumologic.elasticsearch</groupId>
-    <version>1.0.16-SNAPSHOT</version>
+    <version>1.0.17-SNAPSHOT</version>
     <relativePath>../</relativePath>
   </parent>
 
@@ -26,7 +26,7 @@
     <dependency>
       <groupId>com.sumologic.elasticsearch</groupId>
       <artifactId>elasticsearch-test</artifactId>
-      <version>1.0.16-SNAPSHOT</version>
+      <version>1.0.17-SNAPSHOT</version>
       <scope>test</scope>
     </dependency>
 

--- a/elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/AggregationDsl.scala
+++ b/elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/AggregationDsl.scala
@@ -1,4 +1,22 @@
 /**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/**
   * Licensed to the Apache Software Foundation (ASF) under one
   * or more contributor license agreements.  See the NOTICE file
   * distributed with this work for additional information

--- a/elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala
+++ b/elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala
@@ -421,9 +421,7 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
       val expected = BucketAggregationResultBody(0, 0, List(Bucket("aggr1", 1), Bucket("aggr3", 1)))
 
       val aggrQueryFuture = restClient.bucketAggregation(index, tpe, aggrQuery)
-      whenReady(aggrQueryFuture) { resp =>
-        resp should be(expected)
-      }
+      aggrQueryFuture.futureValue should be (expected)
     }
   }
 }

--- a/elasticsearch-test/pom.xml
+++ b/elasticsearch-test/pom.xml
@@ -4,7 +4,7 @@
   <artifactId>elasticsearch-test</artifactId>
   <name>elasticsearch-test</name>
   <packaging>jar</packaging>
-  <version>1.0.16-SNAPSHOT</version>
+  <version>1.0.17-SNAPSHOT</version>
   <scm><connection>scm:git:git@github.com:Sanyaku/sumologic.git/tags/all-1.0.4/sumologic.git</connection></scm>
 
   <properties>
@@ -16,7 +16,7 @@
   <parent>
     <artifactId>all</artifactId>
     <groupId>com.sumologic.elasticsearch</groupId>
-    <version>1.0.16-SNAPSHOT</version>
+    <version>1.0.17-SNAPSHOT</version>
     <relativePath>../</relativePath>
   </parent>
 

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
     <groupId>com.sumologic.elasticsearch</groupId>
     <artifactId>all</artifactId>
     <name>all</name>
-    <version>1.0.16-SNAPSHOT</version>
+    <version>1.0.17-SNAPSHOT</version>
     <packaging>pom</packaging>
     <licenses>
         <license>
@@ -20,8 +20,8 @@
         <module>elasticsearch-aws</module>
     </modules>
     <scm>
-        <url>https://github.com/SumoLogic/elasticsearch-client/tags/all-1.0.16</url>
-        <connection>https://github.com/SumoLogic/elasticsearch-client.git/tags/all-1.0.16</connection>
+        <url>https://github.com/SumoLogic/elasticsearch-client/tags/all-1.0.17</url>
+        <connection>https://github.com/SumoLogic/elasticsearch-client.git/tags/all-1.0.17</connection>
     </scm>
 
     <properties>

